### ISSUE TITLE
ghci ghc-lib-parser load test fails on windows with ghc-8.8.1

### ISSUE THREAD

**Issue by shayne-fletcher-da** [digital-asset/ghc-lib#142]
All Windows flavors (`ghc-8.8.1`, `da-ghc-8.81`, `ghc-master`) compiled with `ghc-8.8.1` fail the load `ghc-lib-parser` in ghci test and are disabled.
```
2019-09-27T22:58:00.4061936Z # Running: stack --resolver nightly-2019-09-26 --no-terminal exec -- ghc -ignore-dot-ghci -package=ghc-lib-parser -e "print 1"
2019-09-27T22:58:00.4758885Z stty: 'standard input': Inappropriate ioctl for device
2019-09-27T22:58:00.7109822Z [0mStack has not been tested with GHC versions above 8.6, and using 8.8.1, this may fail[0m
2019-09-27T22:58:00.8196477Z [0mStack has not been tested with Cabal versions above 2.4, but version 3.0.0.0 was found, this may fail[0m
2019-09-27T22:58:01.6821466Z D:\a\1\s\.stack-work\install\a5e678e4\lib\x86_64-windows-ghc-8.8.1\ghc-lib-parser-0.20190927-FUwiPDktS2BEO3PvZtLNsC\HSghc-lib-parser-0.20190927-FUwiPDktS2BEO3PvZtLNsC.o: unhandled PEi386 relocation type 0
2019-09-27T22:58:01.6853587Z ghc.EXE: unable to load package `ghc-lib-parser-0.20190927'
```

**Comment by mvoidex** [comment]
Same error with `haskell-src-exts-1.21.1` under windows, ghc-8.8.1

**Comment by shayne-fletcher-da** [comment]
Perhaps related :
  - https://ci.appveyor.com/project/ndmitchell/hoogle#L5349
 - https://github.com/ghc/ghc/blob/70e56b272492b65e41a149ec39a939e794fea66b/rts/linker/PEi386.c#L1990
- https://gitlab.haskell.org/ghc/ghc/issues/9907?cversion=0&cnum_hist=5#note_104251

**Comment by shayne-fletcher-da** [comment]
Raised issue with GHC HQ https://gitlab.haskell.org/ghc/ghc/issues/17599.

**Comment by shayne-fletcher-da** [comment]
This issue is said to be an instance of https://gitlab.haskell.org/ghc/ghc/issues/17575 and it is expected that it will be resolved by 8.8.2.

**Comment by flip111** [comment]
> This issue is said to be an instance of

Hi @shayne-fletcher-da on futhark this issue remains with 8.8.2 How do you figure it's the same issue as is fixed in 8.8.2 ?

**Comment by shayne-fletcher** [comment]
HI @flip111 

> > This issue is said to be an instance of
> 
> Hi @shayne-fletcher-da on futhark this issue remains with 8.8.2 How do you figure it's the same issue as is fixed in 8.8.2 ?

It was thought to be so in https://gitlab.haskell.org/ghc/ghc/issues/17599. Just tested - I confirm it's not 😞 

 Perhaps the fix didn't make it into the back-port? Anyway, I'll reopen the above ticket.

**Comment by shayne-fletcher** [comment]
Ticket https://gitlab.haskell.org/ghc/ghc/issues/17599 reopened.

**Comment by shayne-fletcher** [comment]
It appears that indeed the fix didn't make 8.2.2. It is however on the ghc-8.8 branch so will appear in 8.8.3.

**Comment by shayne-fletcher** [comment]
This is confirmed fixed in 8.8.3
